fix(#8837): fix NaN in telemetry doc id

latin-panda opened the pull request:
<!--
Please use semantic PR titles that respect this format:

<type>(#<issue number>): <subject>

Quick example:

feat(#1234): add hat wobble
^--^(#^--^): ^------------^
|     |      |
|     |      + - > subject
|     |
|     + -------- > issue number
|
+ -------------- > type: chore, feat, fix, perf.

https://docs.communityhealthtoolkit.org/contribute/code/workflow/#commit-message-format
-->
# Description

- Adds regex to check the telemetry name
- Adds warning for telemetry names that don't match the new name format.


Testing: https://github.com/medic/cht-core/assets/66472237/436da61e-ef74-4466-8481-66445beb6932

#8837

# Code review checklist
<!-- Remove or comment out any items that do not apply to this PR; in the remaining boxes, replace the [ ] with [x]. -->
- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.






# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-couchdb-clustered.yml)


# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

latin-panda commented:
@garethbowen @ralfudx I ran the v4.2.x and 4.5.x branches, and after several tries, I couldn't get a bad telemetry record. I straightened the validations and added an exception, so next time, we will have more info from the feedback docs. 
Let me know your suggestions. Thanks!

garethbowen commented:
@latin-panda Thanks for getting right on this!

To do a review I need more information, like how you determined this is the problem, and why this is the right fix. The best place for this is probably a comment on the issue.

latin-panda commented:
>  how you determined this is the problem

@garethbowen I haven't been able to reproduce it. So maybe @ralfudx can help? 
For now, this is just making the validations stronger by processing the telemetry db names with the correct format. 
I'll add this in the ticket in a couple of minutes

garethbowen commented on webapp/src/ts/services/telemetry.service.ts:
```diff
@@ -345,10 +356,8 @@ export class TelemetryService {
-      // Skips new Telemetry DB, then matches the old deprecated Telemetry DB.
-      if (!nameNoPrefix.startsWith(this.TELEMETRY_PREFIX)
-        && nameNoPrefix.includes(this.TELEMETRY_PREFIX)
-        && nameNoPrefix.includes(this.sessionService.userCtx().name)) {
```
Why did you remove the check for `userCtx`? Without this won't we end up deleting someone elses telemetry on phones that are shared between users?

latin-panda replied:
ah shared phones, okay putting it back

garethbowen commented:
@latin-panda Sorry I wrote that comment before you updated the PR description and added the comment. I deleted it again once I saw the new details. Sorry for the confusion!

garethbowen reviewed (changes requested):
I worry this is going to hide the error, because now any database that doesn't match the name will be deleted. In other words, if this happens again it'll silently fail rather than loudly fail. Because we haven't managed to reproduce this yet, we can't be certain if deletion is the right approach or if there's a bug in how we're creating the dbs in the first place.

Let's put some more time into trying to reproduce it.

latin-panda commented:
@ralfudx and @garethbowen please have a look. The momentjs format was translating the numbers to Nepali, I have updated the ticket with this.  

@ralfudx Please check the PR description, I have added a video showing how to test this error. Let me know any questions or suggestions.

garethbowen reviewed (approved):
LGTM. Thanks for adding the bikram sambat test!

ralfudx reviewed (approved):
LGTM
